src/pipes/search.ts

```diff
@@ -0,0 +1,18 @@
+import { Pipe } from '@angular/core';
+
+@Pipe({
+  name: 'search'
+})
+export class SearchPipe {
+  transform(list, pattern = '', deepKey = '') {
+    if (!list) return;
+
+    return list.filter((item) => {
+      const value = deepKey.split('.').reduce((value, key) => {
```

Comment:
`lodash` is a huge ass library and I'm not willing to load of it just for a simple poor utility function. If you want I can install a dedicated module.